src/pages/en/guides/cms/contentful.md

```diff
@@ -0,0 +1,535 @@
+
+```astro title="src/pages/index.astro" ins={9-17}
+---
+import { contentfulClient } from "../lib/contentful";
+import type { blogPost } from "../lib/contentful";
+
+const entries = await contentfulClient.getEntries<blogPost>({
+  content_type: "blogPost",
+});
+
+const posts = entries.items.map((item) => {
+  const { title, date, description, slug } = item.fields;
+  return {
+    title,
+    slug,
+    description,
+    date: new Date(date).toLocaleDateString()
+  };
+});
+---
+```
+
+Finally, you can use the `posts` variable to write the markup of your blog posts previews.
+
+```astro astro title="src/pages/index.astro" ins={19-37}
+---
+import { contentfulClient } from "../lib/contentful";
+import type { blogPost } from "../lib/contentful";
+
+const entries = await contentfulClient.getEntries<blogPost>({
+  content_type: "blogPost",
+});
+
+const posts = entries.items.map((item) => {
+  const { title, date, description, slug } = item.fields;
+  return {
+    title,
+    slug,
+    description,
+    date: new Date(date).toLocaleDateString()
+  };
+});
+---
+<html lang="en">
+  <head>
+    <title>My Blog</title>
+  </head>
+  <body>
+    <h1>My Blog</h1>
+    <ul>
+      {posts.map((post) => (
+        <li>
+          <a href={`/posts/${slug}/`}>
+            <h2>{title}</h2>
+          </a>
+          <time>{date}</time>
+          <p>{description}</p>
+        </li>
+      ))}
+    </ul>
+  </body>
+</html>
+```
+
+### Dynamic routes
+
+#### Static site generation
+
+To generate our blog posts pages statically, we will use [dynamic routes](/en/core-concepts/routing/#dynamic-routes) and the `getStaticPaths()` function. This function will be called at build time to generate the list of paths that will be rendered to HTML.
+
+Create a new file named `[slug].astro` in `src/pages/posts/` and import the `blogPost` interface and `contentfulClient` from `src/lib/contentful.ts`. 
+
+Inside the `getStaticPaths()` function, fetch all the entries with a content type of `blogPost` using the `getEntries` method from `contentfulClient`. You can pass the `blogPost` interface to the `getEntries` method to type your response. 
+
+```astro title="src/pages/posts/[slug].astro"
+---
+import { contentfulClient } from "../../lib/contentful";
+import type { blogPostFields } from "../../lib/contentful";
+
+export async function getStaticPaths() {
+  const entries = await contentfulClient.getEntries<blogPost>({
+    content_type: "blogPost",
+  });
+}
+---
+```
+
+The items property of `entries` contains a list of blog posts. We will use the `map` method to iterate over the list and return a list of pages. Each page will be an object with a `params` and `props` property. The `params` property will be used to generate the URL of the page and the `props` property will be passed to the page component as props.
```

Comment:
This paragraph mostly describes the code (rather than explaining a deeper concept behind it), and it's harder to follow without looking at the code, so I think it's worth shortening it:
```suggestion
Then, map each item to an object with a `params` and `props` property. The `params` property will be used to generate the URL of the page and the `props` property will be passed to the page component as props.
```